packages/utils/src/logger.ts

```diff
@@ -1,118 +1,97 @@
-/* eslint-disable @typescript-eslint/no-explicit-any */
 import { WrappedFunction } from '@sentry/types';
 
 import { IS_DEBUG_BUILD } from './flags';
-import { getGlobalObject } from './global';
+import { getGlobalObject, getGlobalSingleton } from './global';
 
 // TODO: Implement different loggers for different environments
 const global = getGlobalObject<Window | NodeJS.Global>();
 
 /** Prefix for logging strings */
 const PREFIX = 'Sentry Logger ';
 
-export const CONSOLE_LEVELS = ['debug', 'info', 'warn', 'error', 'log', 'assert'];
+export const CONSOLE_LEVELS = ['debug', 'info', 'warn', 'error', 'log', 'assert'] as const;
+
+type LoggerMethod = (...args: unknown[]) => void;
+type LoggerConsoleMethods = Record<typeof CONSOLE_LEVELS[number], LoggerMethod>;
 
 /** JSDoc */
-interface ExtensibleConsole extends Console {
-  [key: string]: any;
+interface Logger extends LoggerConsoleMethods {
+  disable(): void;
+  enable(): void;
 }
 
 /**
- * Temporarily unwrap `console.log` and friends in order to perform the given callback using the original methods.
- * Restores wrapping after the callback completes.
+ * Temporarily disable sentry console instrumentations.
  *
  * @param callback The function to run against the original `console` messages
  * @returns The results of the callback
  */
-export function consoleSandbox(callback: () => any): any {
+export function consoleSandbox<T>(callback: () => T): T {
```

Comment:
as a next step, we can further refactor `consoleSandbox` to take a `level` argument, so we don't have to un-patch and re-patch every single console level every time we call this.